src/drive/mobile/modules/mediaBackup/duck/index.js

```diff
@@ -34,9 +37,95 @@ const currentMediaUpload = (media, uploadCounter, totalUpload) => ({
   }
 })
 
-const getDirID = async path => {
-  const { _id } = await cozy.client.files.createDirectoryByPath(path, false)
-  return _id
+const getFoldersWithReference = async reference => {
+  const referencedFolders = await cozy.client.fetchJSON(
+    'POST',
+    '/files/_find',
+    {
+      selector: {
+        referenced_by: {
+          $elemMatch: {
+            type: reference
+          }
+        }
+      },
+      sort: [{ created_at: 'desc' }]
+    }
+  )
+
+  return referencedFolders.filter(
+    folder => /^\/\.cozy_trash/.test(folder.attributes.path) === false
+  )
+}
+
+const addFolderReference = (folderId, reference) =>
+  cozy.client.fetchJSON(
+    'POST',
+    `/files/${folderId}/relationships/referenced_by`,
+    {
+      data: [
+        {
+          type: reference,
+          id: reference
+        }
+      ]
+    }
+  )
+
+const ensureFolderWithReferenceExists = async (path, reference) => {
+  const referencedFolders = await getFoldersWithReference(reference)
+
+  if (referencedFolders.length) {
+    return referencedFolders[0]
+  } else {
+    const dir = await cozy.client.files.createDirectoryByPath(path, false)
+    await addFolderReference(dir._id, reference)
+    return dir
+  }
+}
+
+const ensureHasCreatedAtIndex = () =>
+  cozy.client.fetchJSON('POST', '/data/io.cozy.files/_index', {
+    index: {
+      fields: ['created_at']
+    }
+  })
+
+const getUploadDir = async () => {
+  await ensureHasCreatedAtIndex()
+
+  const uploadedFolders = await getFoldersWithReference(REFERENCE_UPLOAD_DIR)
+
+  if (uploadedFolders.length >= 1) {
+    return uploadedFolders[0]
+  } else {
+    const t = getTranslateFunction()
+    const mediaFolderName = t('mobile.settings.media_backup.media_folder')
+    const uploadFolderName = t('mobile.settings.media_backup.backup_folder')
+    const legacyUploadFolderName = t(
+      'mobile.settings.media_backup.legacy_backup_folder'
+    )
+
+    await ensureFolderWithReferenceExists(
+      `/${mediaFolderName}`,
+      REFERENCE_PHOTOS_DIR
+    )
```

Comment:
Why this one is not in the `try` block ?

Reply:
It is, [in the function above](https://github.com/y-lohse/cozy-drive/blob/magic-folders/src/drive/mobile/modules/mediaBackup/duck/index.js#L160) ;)

Unless you see something useful we could do here in case we can't create the `Photos` folder?